cmd/snap/cmd_run.go

```diff
@@ -95,9 +97,54 @@ func (x *cmdRun) Execute(args []string) error {
 		x.Command = "shell"
 	}
 
+	if x.Command == "complete" {
+		snapApp, args = antialias(snapApp, args)
+	}
+
 	return snapRunApp(snapApp, x.Command, args)
 }
 
+// antialias changes snapApp and args if snapApp is actually an alias
+// for something else. If not, or if the args aren't what's expected
+// for completion, it returns them unchanged.
+func antialias(snapApp string, args []string) (string, []string) {
+	if len(args) < 7 {
+		// NOTE if len(args) < 7, Something is Wrong (at least WRT complete.sh and etelpmoc.sh)
+		return snapApp, args
+	}
+
+	actualApp, err := resolveApp(snapApp)
+	if err != nil || actualApp == snapApp {
+		// no alias! woop.
+		return snapApp, args
+	}
+
+	compPoint, err := strconv.Atoi(args[2])
+	if err != nil {
+		// args[2] is not COMP_POINT?
+		return snapApp, args
+	}
+
+	if args[6] != snapApp {
+		// args[6] is not COMP_WORDS[0]?
+		return snapApp, args
+	}
+
+	re := regexp.MustCompile(`^` + regexp.QuoteMeta(snapApp) + `\b`)
```

Comment:
well, the point of QuoteMeta is to make it sane no matter what, so if it panics it would be a bug in regexp itself afaiu, but I can also see a point of making it a normal error

Reply:
it wouldn't hurt to bail on error instead of panic'ing, indeed.

There's no space for 'normal error' here unfortunately